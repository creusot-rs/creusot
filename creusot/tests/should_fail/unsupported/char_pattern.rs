extern crate creusot_contracts;

pub fn char_match_bad(x: char) {
    match x {
        'a' => (),
        _ => (),
    }
}
