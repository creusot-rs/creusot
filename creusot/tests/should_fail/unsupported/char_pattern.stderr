warning: support for string types is limited and experimental
 --> char_pattern.rs:4:11
  |
4 |     match x {
  |           ^
  |
  = note: `#[warn(creusot::experimental)]` on by default

warning: support for string types is limited and experimental
 --> char_pattern.rs:5:9
  |
5 |         'a' => (),
  |         ^^^

error: match on char is currently unsupported
 --> char_pattern.rs:4:5
  |
4 |     match x {
  |     ^^^^^^^

error: aborting due to 1 previous error; 2 warnings emitted

